armi/utils/hexagon.py

```diff
@@ -145,3 +145,50 @@ def numPositionsInRing(ring):
         rings is indexed to 1, i.e. the centermost position in the lattice is ``ring=1``.
     """
     return (ring - 1) * 6 if ring != 1 else 1
+
+
+def totalPositionUpToRing(ring: int) -> int:
+    """Return the number of positions in a hexagon with a given number of rings."""
+    return 1 + 3 * ring * (ring - 1)
+
+
+def getIndexOfRotatedCell(initialCellIndex: int, orientationNumber: int) -> int:
```

Comment:
This could also use some direct unit testing.  For instance:

* Test a `self.assertRaises(ValueError)` for invalid orientation number.
* Test a couple known results for when `initialCellIndex > 1`.
* Test a couple known results for when `initialCellIndex== 1`.
* Test a `self.assertRaises(ValueError)` for negative cell number.